docs/proposals/mapper- provide-a- common- MQTT- mapper.md

```diff
@@ -0,0 +1,277 @@
+
+### Proposal
+
+        Providing a common MQTT mapper using the latest mapper-framework aims to help developers easily make use of mapper to solve MQTT communication, in order to simplify the device integration process, improve system flexibility and reliability.
+
+##### Scope
+
+- Development of the MQTT Mapper: Utilize the latest Mapper-Framework to add an MQTT protocol Mapper to the kubeedge/mappers-go project. Ensure the Mapper can subscribe to specified MQTT topics, receive messages, and parse them.
+
+- Message Parsing and Serialization Support: functionality to retrieve messages from MQTT service topics. Support parsing messages using JSON, YAML and XML formats to extract attribute values.
+
+##### User
+
+- Developers: Developers are needed to implement and test the MQTT Mapper.
+
+- IoT Device Manufacturers and Operators: IoT device manufacturers and operators can use this universal MQTT Mapper to manage device communications.
+
+### Design Details
+
+##### Architecture and Modules
+
+        The architecture for Mapper Provide a Common Mapper will be modular to ensure scalability and maintainability. The main components include:
+
+- Base Environment Setup: Utilizes the latest mapper-framework to create a MQTT mapper for standardized operatons and interactions.
+
+- Main algorithm design: Parses the attribute values from the message using common serialization methods(JSON, YAML, XML).
+
+        The architectures and related concepts are shown in the below figure. In the figure, we assume that in the cloud-side-end scenario, device state modifications can be initiated by the cloud.
+
+<img title="相对路径演示" src="../images/proposals/mapper-provide-a-common-MQTT-mapper-architecture.jpg" alt="本地路径" width="546" data-align="center">
+
+##### Base Environment Setup
+
+1. Generate Mapper files through the Mapper-Framework framework
+   
+   1)Download kubeedge-master.zip from the official website, take the mapper-framewokr under the staging folder in it, and transfer it to the VM master node.
+   
+   2)Execute a command: 
+   
+   ```go
+   make generate
+   ```
+   
+   3)Enter the project name of the Mapper)
+   
+   <img title="" src="../images/proposals/mapper-provide-a-common-MQTT-mapper-pic.jpg" alt="" data-align="center">
+
+2. Determine the Device model and Device instance files for device
+   
+   1)Device model: A device model describes the device properties exposed by the device . A device model is a Physical model which constrains the properties and parameters of physical devices.
+   
+   ```
+   apiVersion: devices.kubeedge.io/v1beta1
+   kind: DeviceModel
+   metadata:
+     name: beta1-model
+   spec:
+     properties:
+       - name: temp
+         description: beta1-model
+         type: INT
+         accessMode: ReadWrite
+         maximum: "100"
+         minimum: "1"
+         unit: "Celsius"
+     protocol: modbus
```

Comment:
```suggestion
     protocol: mqtt
```

This proposal is to provide a mapper for the mqtt protocol, so the protocol name here should be mqtt

Reply:
Ok, I'll all change the model and instance files based on your comments.